XiEditor/AppDelegate.swift

```diff
@@ -262,6 +203,21 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         }
     }
 
+    //MARK: helpers
+
+    /// returns the NSDocument corresponding to the given viewIdentifier
+    private func documentForViewIdentifier(viewIdentifier: ViewIdentifier) -> Document? {
+        //TODO: move this to use a hashmap; we will have to do some manual bookkeeping,
+        // but this is a noticeable (small) bottleneck in some scenarios
```

Comment:
I wouldn't consider it small necessarily, but no need to quibble. On the other hand, I think it _is_ good to point out that this method needs to be thread-safe (as it's called on the update method from the read thread), even if the current impl isn't doing anything special to respect that.